compiler/rustc_codegen_llvm/src/gotoc/cbmc/goto_program/symtab_transformer/transformer.rs

```diff
@@ -0,0 +1,742 @@
+            Type::Bool => self.transform_bool_type(),
+            Type::CBitField { typ, width } => self.transform_c_bit_field_type(typ, width),
+            Type::CInteger(c_int_type) => self.transform_c_integer_type(c_int_type),
+            Type::Code { parameters, return_type } => {
+                self.transform_code_type(parameters, return_type)
+            }
+            Type::Constructor => self.transform_constructor_type(),
+            Type::Double => self.transform_double_type(),
+            Type::Empty => self.transform_empty_type(),
+            Type::FlexibleArray { typ } => self.transform_flexible_array_type(typ),
+            Type::Float => self.transform_float_type(),
+            Type::IncompleteStruct { tag } => self.transform_incomplete_struct_type(tag),
+            Type::IncompleteUnion { tag } => self.transform_incomplete_union_type(tag),
+            Type::InfiniteArray { typ } => self.transform_infinite_array_type(typ),
+            Type::Pointer { typ } => self.transform_pointer_type(typ),
+            Type::Signedbv { width } => self.transform_signedbv_type(width),
+            Type::Struct { tag, components } => self.transform_struct_type(tag, components),
+            Type::StructTag(tag) => self.transform_struct_tag_type(tag),
+            Type::Union { tag, components } => self.transform_union_type(tag, components),
+            Type::UnionTag(tag) => self.transform_union_tag_type(tag),
+            Type::Unsignedbv { width } => self.transform_unsignedbv_type(width),
+            Type::VariadicCode { parameters, return_type } => {
+                self.transform_variadic_code_type(parameters, return_type)
+            }
+            Type::Vector { typ, size } => self.transform_vector_type(typ, size),
+        }
+    }
+
+    /// Transforms an array type (`typ x[size]`)
+    fn transform_array_type(&self, typ: &Box<Type>, size: &u64) -> Type {
+        let transformed_typ = self.transform_type(typ.as_ref());
+        transformed_typ.array_of(*size)
+    }
+
+    /// Transforms a CPROVER boolean type (`__CPROVER_bool x`)
+    fn transform_bool_type(&self) -> Type {
+        Type::bool()
+    }
+
+    /// Transforms a c bit field type (`typ x : width`)
+    fn transform_c_bit_field_type(&self, typ: &Box<Type>, width: &u64) -> Type {
+        let transformed_typ = self.transform_type(typ.as_ref());
+        transformed_typ.as_bitfield(*width)
+    }
+
+    /// Transforms a machine-dependent integer type (`bool`, `char`, `int`, `size_t`)
+    fn transform_c_integer_type(&self, c_int_type: &CIntType) -> Type {
+        match c_int_type {
+            CIntType::Bool => Type::c_bool(),
+            CIntType::Char => Type::c_char(),
+            CIntType::Int => Type::c_int(),
+            CIntType::SizeT => Type::size_t(),
+            CIntType::SSizeT => Type::ssize_t(),
+        }
+    }
+
+    /// Transforms a parameter for a function
+    fn transform_parameter(&self, parameter: &Parameter) -> Parameter {
```

Comment:
Shouldn't this be `transform_parameter_type` to match the other type-transforming function names?

Reply:
It felt odd to give it that name since the parameter itself isn't a type (it's a data contained within some types), but for organization purposes that would makes sense.

Reply:
Fixed!